cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1717,6 +1717,12 @@ const FunctionDoc pow_doc{
      "wraps around. If either base or exponent is null the result will be null."),
     {"base", "exponent"}};
 
+const FunctionDoc exp_doc{
+    "Computes Euler's number raised to the specified power, element-wise",
+    ("Negative integer power returns an error. However, integer overflow\n"
```

Comment:
```suggestion
    ("Negative integer power returns an error. However, integer overflow\n"
```
What does negative integer power mean?  This function only has non-integral kernels.

I'm not sure what you mean by integer overflow.  Why "integer"?  The return value is a floating point.  If there is an overflow then "[+HUGE_VAL](https://en.cppreference.com/w/cpp/numeric/math/HUGE_VAL), +HUGE_VALF, or +HUGE_VALL is returned."

Reply:
That was a placeholder from power. I updated it.